quic/s2n-quic-transport/src/connection/connection_impl.rs

```diff
@@ -1049,7 +1049,27 @@ impl<Config: endpoint::Config> connection::Trait for ConnectionImpl<Config> {
     }
 
     /// Process ACKs for the `Connection`.
-    fn on_process_acks(&mut self) {}
+    fn on_pending_ack_ranges(
+        &mut self,
+        timestamp: Timestamp,
+        subscriber: &mut Config::EventSubscriber,
+    ) {
+        let mut publisher = self.event_context.publisher(timestamp, subscriber);
+
+        // TODO: care should be taken to only delay ACK processing for the active path.
+        // However, the active path could change so it might be necessary to track the
+        // active path across some ACK delay processing.
+        let path_id = self.path_manager.active_path_id();
+        if let Err(_err) = self.space_manager.on_pending_ack_ranges(
+            timestamp,
+            path_id,
+            &mut self.path_manager,
+            &mut self.local_id_registry,
+            &mut publisher,
+        ) {
+            // TODO: publish metrics
```

Comment:
We need to do more than just publish metrics. The recovery manager could decide to return a connection error. For example, if the peer is ACKing a packet number that we haven't sent yet is a protocol violation.

I think the `on_pending_ack_ranges` should return a `Result<(), connection::Error>`

Reply:
yep, protocol violations need to bubble up. I wanted this PR to focus on the interface so I am going to comment out the usage in endpoint/mod and tackle that as part of the following PR.

This function will return `Result`

Reply:
Returning the error in https://github.com/aws/s2n-quic/pull/1293/commits/9d0dd0e123aa9dd6335648069baba8517eb9253d

commented out the usage in endpoint/mod till next PR and punt on connection close